tasks/eth/006-MCP-L1/VALIDATION.md

```diff
@@ -0,0 +1,335 @@
+      - The location (`8`) of the `approvedHashes` mapping in the [GnosisSafe storage layout](https://github.com/safe-global/safe-contracts/blob/v1.4.0/contracts/libraries/SafeStorage.sol#L23)
+      - The address of the Council Safe: `0xc2819DC788505Aac350142A7A707BF9D03E3Bd03`
+      - The safe hash to approve: `0x3ac2526a656ea43a0a7e3926d44768d54616d8370fdaf5538390bba29f31b245`
+    - The using `cast index`, we can verify that:
+      ```shell
+        $ cast index address 0xc2819DC788505Aac350142A7A707BF9D03E3Bd03 8
+        0xaaf2b641eaf0bae063c4f2e5670f905e1fb7334436b902d1d880b05bd6228fbd
+        ```
+        and
+      ```shell
+        $ cast index bytes32 0x3ac2526a656ea43a0a7e3926d44768d54616d8370fdaf5538390bba29f31b245 0xaaf2b641eaf0bae063c4f2e5670f905e1fb7334436b902d1d880b05bd6228fbd
+        0xe5a48f5ffaa688c5a5bcf794e48eb875102b7174501dea10998e539cfda7b3d1
+        ```
+      And so the output of the second command matches the key above.
+
+#### For the Foundation:
+
+- **Key:** `0x8bbe2c1d1398aac81019c2c136ded3baed46d82f19927d40eb4f0c11b4cd4744` <br/>
+  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000000`<br/>
+  **After:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>
+  **Meaning:** The GnosisSafe `approvedHashes` mapping is updated to indicate approval of this transaction by the council. The correctness of this slot can be verified as follows:
+    - Since this is a nested mapping, we need to use `cast index` twice to confirm that this is the correct slot. The inputs needed are:
+      - The location (`8`) of the `approvedHashes` mapping in the [GnosisSafe storage layout](https://github.com/safe-global/safe-contracts/blob/v1.4.0/contracts/libraries/SafeStorage.sol#L23)
+      - The address of the Council Safe: `0x5a0aae59d09fccbddb6c6cceb07b7279367c3d2a`
+      - The safe hash to approve: `0x3ac2526a656ea43a0a7e3926d44768d54616d8370fdaf5538390bba29f31b245`
+    - The using `cast index`, we can verify that:
+      ```shell
+        $ cast index address 0x847B5c174615B1B7fDF770882256e2D3E95b9D92 8
+        0x13908ba1c0e379ab58c6445554ab471f3d4efb06e3c4cf966c4f5e918eca67bd
+      ```
+      and
+      ```shell
+        $ cast index bytes32 0x3ac2526a656ea43a0a7e3926d44768d54616d8370fdaf5538390bba29f31b245 0x13908ba1c0e379ab58c6445554ab471f3d4efb06e3c4cf966c4f5e918eca67bd
+        0x8bbe2c1d1398aac81019c2c136ded3baed46d82f19927d40eb4f0c11b4cd4744
+      ```
+      And so the output of the second command matches the key above.
+
+### `0x5a7749f83b81b301cab5f48eb8516b986daef23d` (`L1ERC721BridgeProxy`)
+
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>
+  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>
+  **After:** `0x00000000000000000000000025ace71c97b33cc4729cf772ae268934f7ab5fa1` <br/>
+  **Meaning:** The `initialized` boolean is set to `true`.
```

Comment:
This is setting the L1CrossDomainMessengerProxy in slot 1. Should there also be a slot 0 change for the initializer?

Reply:
just fixed the description for the changes here, but looks like initializer is not set in this case. i'd assume that it's bc we are reusing the proxy and the slot there is already initialized?

Reply:
The initializer does not show because it goes from 1 to 0 to 1 during the transaction, but from the sim we can see the Initialized events are emitted